### PR TITLE
[testing-on-gke part 6.2] extract common runner code

### PR DESCRIPTION
**Pull request by gargnitingoogle**
### Description
It extracts common code from fio/run_tests.py and dlio/run_tests.py and puts it in utils/run_tests_common.py for sharing.

This is on top of #2489 and is followed in #2495 .

### Link to the issue in case of a bug fix.
NA

### Testing details
1. Manual - NA
2. Unit tests - NA
3. Integration tests - NA

### CODE REVIEW

File: perfmetrics/scripts/testing_on_gke/examples/utils/run_tests_common_test.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  def test_escape_commas_in_string(self):
+    tcs = [
+        {"input": "a:b,c=d,", "expected_output": "a:b\,c=d\,"},
+        {"input": "", "expected_output": ""},
+    ]
+    for tc in tcs:
+      self.assertEqual(
+          tc["expected_output"], escape_commas_in_string(tc["input"])
+      )
+      pass
```
Comment (kislaykishore) [inline perfmetrics/scripts/testing_on_gke/examples/utils/run_tests_common_test.py:36]: "pass" is redundant.
```suggestion
```

File: perfmetrics/scripts/testing_on_gke/examples/utils/run_tests_common_test.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import unittest
+from run_tests_common import escape_commas_in_string
+# import utils
+
+
+class RunTestsCommonTest(unittest.TestCase):
+
+  @classmethod
+  def setUpClass(self):
+    pass
```
Comment (kislaykishore) [inline perfmetrics/scripts/testing_on_gke/examples/utils/run_tests_common_test.py:25]: Remove no-op code.
```suggestion
```

File: perfmetrics/scripts/testing_on_gke/examples/utils/run_tests_common.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+      ),
+  )
+
+  args = parser.parse_args()
+  for argument in [
+      'instance_id',
+      'machine_type',
+  ]:
+    value = getattr(args, argument)
+    if len(value) == 0 or str.isspace(value):
```
Comment (kislaykishore) [inline perfmetrics/scripts/testing_on_gke/examples/utils/run_tests_common.py:89]:
```suggestion
    if value.strip():
```

File: perfmetrics/scripts/testing_on_gke/examples/utils/run_tests_common_test.py (hunk 1 above)
Comment (gargnitingoogle) [reply perfmetrics/scripts/testing_on_gke/examples/utils/run_tests_common_test.py]: done

File: perfmetrics/scripts/testing_on_gke/examples/utils/run_tests_common_test.py (hunk 2 above)
Comment (gargnitingoogle) [reply perfmetrics/scripts/testing_on_gke/examples/utils/run_tests_common_test.py]: done

File: perfmetrics/scripts/testing_on_gke/examples/utils/run_tests_common.py (hunk 3 above)
Comment (gargnitingoogle) [reply perfmetrics/scripts/testing_on_gke/examples/utils/run_tests_common.py]: done

**Review by kislaykishore — APPROVED** [review]

**Pull request merged**